### PR TITLE
New squad example

### PR DESCRIPTION
**Pull request by sgugger**
Reopening from #8924 since the rebase gave too big a diff.

### CODE REVIEW

File: examples/question-answering/run_qa.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+        validation_dataset = datasets["validation"].map(
+            prepare_validation_features,
+            batched=True,
+            num_proc=data_args.preprocessing_num_workers,
+            remove_columns=column_names,
+            load_from_cache_file=not data_args.overwrite_cache,
+        )
+
+    # Data collator
+    data_collator = default_data_collator if data_args.pad_to_max_length else DataCollatorWithPadding(tokenizer)
```
Comment (LysandreJik) [inline examples/question-answering/run_qa.py:391]: This seems a bit unintuitive, I would expect the `DataCollatorWithPadding` data collator to be used if we want to pad to max length

File: examples/question-answering/run_qa_beam_search.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+        validation_dataset = datasets["validation"].map(
+            prepare_validation_features,
+            batched=True,
+            num_proc=data_args.preprocessing_num_workers,
+            remove_columns=column_names,
+            load_from_cache_file=not data_args.overwrite_cache,
+        )
+
+    # Data collator
+    data_collator = default_data_collator if data_args.pad_to_max_length else DataCollatorWithPadding(tokenizer)
```
Comment (LysandreJik) [inline examples/question-answering/run_qa_beam_search.py:432]: Same here

**Review by LysandreJik — APPROVED** [review]
This looks great!

File: examples/question-answering/run_qa.py (hunk 1 above)
Comment (sgugger) [reply examples/question-answering/run_qa.py:391]: Will had a comment.

Comment (LysandreJik) [reply examples/question-answering/run_qa.py]: Looking forward to Will's comment

**Pull request merged**